### PR TITLE
Feature/test notifications

### PR DESCRIPTION
**Pull request by draskonovakovic**

### CODE REVIEW

File: Konteh/back-office-app/src/app/modules/exam/started-exams/started-exams.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (ukinolo) [inline Konteh/back-office-app/src/app/modules/exam/started-exams/started-exams.component.spec.ts:1]: delete spec.ts

File: Konteh/back-office-app/src/app/app.module.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
-import { NgModule } from '@angular/core';
+import { APP_INITIALIZER, NgModule } from '@angular/core';
```
Comment (ukinolo) [inline Konteh/back-office-app/src/app/app.module.ts:1]: remove not used imports

**Comment by laslorajda** [comment]
you have merge conflicts

File: Konteh/Konteh.FrontOffice.Api/Konteh.FrontOffice.Api.csproj [hunk 3]
```diff
@@ -7,6 +7,10 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="MassTransit" Version="8.2.4" />
+    <PackageReference Include="MassTransit.RabbitMQ" Version="8.2.4" />
+    <PackageReference Include="MediatR" Version="12.4.0" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR" Version="1.1.0" />
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOffice.Api/Konteh.FrontOffice.Api.csproj:13]: Why is SignalR part of FrontOffice.Api?

File: Konteh/Konteh.Infrastructure/Konteh.Infrastructure.csproj [hunk 4]
```diff
@@ -8,6 +8,9 @@
     </PropertyGroup>
 
     <ItemGroup>
+      <PackageReference Include="MassTransit" Version="8.2.4" />
+      <PackageReference Include="MassTransit.RabbitMQ" Version="8.2.4" />
```
Comment (laslorajda) [inline Konteh/Konteh.Infrastructure/Konteh.Infrastructure.csproj:12]: Do we need these packages in infra?

File: Konteh/back-office-app/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -1,11 +1,20 @@
...
 
 const routes: Routes = [
   {
     path: "questions/overview",
     component: OverviewComponent,
+    canActivate: [AuthGuard],
+  },
+  {
+    path: "exams/notifications",
+    component: StartedExamsComponent,
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/app-routing.module.ts:16]: I'd probably delete this component, it offers no value.

File: Konteh/back-office-app/src/app/signalr.service.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+export class SignalRService {
+  private hubConnection: signalR.HubConnection | undefined;
+  private message = new Subject<string>();
+  public message$ = this.message.asObservable();
+
+  constructor() { }
+
+  public async startConnection() {
+    this.hubConnection = new signalR.HubConnectionBuilder()
+      .withUrl('https://localhost:7127/notifications')
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/signalr.service.ts:17]: Please make this configurable

File: Konteh/back-office-app/src/environments/environment.development.ts [hunk 7]
```diff
@@ -3,7 +3,7 @@ export const environment = {
     msalConfig: {
         clientId: '4e1ff54b-bf34-4f45-83ce-e50fc32967cd',
         authority: 'https://login.microsoftonline.com/common',
-        redirectUri: 'http://localhost:4200',
+        redirectUri: 'http://localhost:4401',
```
Comment (laslorajda) [inline Konteh/back-office-app/src/environments/environment.development.ts:6]: I keep seeing this port all over the place. First off it should be configurable, but more importantly, I don't see any guarantees that the web app will be running on this port?

File: Konteh/front-office-app/src/app/app-routing.module.ts [hunk 8]
```diff
@@ -1,7 +1,15 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { StartExamComponent } from './modules/exams/start-exam/start-exam.component';
+import { AuthGuard } from './authentication/auth.guard';
 
-const routes: Routes = [];
+const routes: Routes = [
+{
+  path: "exams/start-exam",
+  component: StartExamComponent,
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/app-routing.module.ts:9]: again, I would probably remove this component, it was fine to have it while you were testing, but now it does nothing.

File: Konteh/front-office-app/src/app/app.component.ts [hunk 9]
```diff
@@ -1,10 +1,47 @@
...
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrl: './app.component.css'
 })
 export class AppComponent {
-  title = 'front-office-app';
+  title = 'back-office-app';
+  loginDisplay = false;
+
+  constructor(private authService: MsalService,private router: Router) { }
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/app.component.ts:17]: Remove MSAL from frontoffice, not sure why it was added to begin with.

File: Konteh/Konteh.FrontOffice.Api/Controllers/ExamController.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+public class ExamController : ControllerBase
+{
+    private readonly IPublishEndpoint _publishEndpoint;
+
+    public ExamController(IPublishEndpoint publishEndpoint)
+    {
+        _publishEndpoint = publishEndpoint;
+    }
+
+    [HttpPost("{message}")]
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOffice.Api/Controllers/ExamController.cs:18]: I wouldn't have this as part of the route in a post method. In Post methods you put stuff in the request body.

File: Konteh/Konteh.FrontOffice.Api/Program.cs [hunk 11]
```diff
@@ -8,6 +12,42 @@
 WebApplicationBuilder? builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
+builder.Logging.ClearProviders();
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOffice.Api/Program.cs:15]: what now?

File: Konteh/Konteh.FrontOffice.Api/Program.cs [hunk 12]
```diff
@@ -8,6 +12,42 @@
...
+builder.Logging.AddConsole();
+
+// Add services to the container.
+builder.Services.AddAuthentication(options =>
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOffice.Api/Program.cs:19]: Why is this still here?

File: Konteh/back-office-app/src/app/app-routing.module.ts (hunk 5 above)
Comment (laslorajda) [reply Konteh/back-office-app/src/app/app-routing.module.ts:16]: I still see this component?

File: Konteh/back-office-app/src/app/app.component.html [hunk 13]
```diff
@@ -3,7 +3,7 @@
 
   <div class="toolbar-spacer"></div>
 
-  <button *ngIf="loginDisplay" (click)="test123()">Test123</button>
+  <button mat-raised-button *ngIf="examNotificationsDisplay" (click)="examNotifications()">Exam Notifications</button>
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/app.component.html:6]: delete

File: Konteh/back-office-app/src/app/modules/initial/test/test.component.css
Comment (laslorajda) [inline Konteh/back-office-app/src/app/modules/initial/test/test.component.css:1]: what is this empty file here for?

File: Konteh/front-office-app/src/app/authentication/auth.guard.ts [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from "@angular/core";
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router } from "@angular/router";
+import { MsalService } from "@azure/msal-angular";
+
+@Injectable({
+    providedIn: "root",
+})
+export class AuthGuard implements CanActivate {
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/authentication/auth.guard.ts:8]: I thought we said we would be deleting msal from frontoffice?

File: Konteh/back-office-app/src/environments/environment.development.ts (hunk 7 above)
Comment (laslorajda) [reply Konteh/back-office-app/src/environments/environment.development.ts]: I don't see an answer to the question.

File: Konteh/Konteh.Domain/Konteh.Domain.csproj [hunk 15]
```diff
@@ -7,4 +7,10 @@
         <Nullable>enable</Nullable>
     </PropertyGroup>
 
+    <ItemGroup>
```
Comment (laslorajda) [inline Konteh/Konteh.Domain/Konteh.Domain.csproj:10]: Why is this still here?

File: Konteh/front-office-app/src/app/app.module.ts [hunk 16]
```diff
@@ -1,20 +1,32 @@
...
 
 @NgModule({
   declarations: [
     AppComponent
   ],
   imports: [
     BrowserModule,
-    AppRoutingModule
+    AppRoutingModule,
+    MatButtonModule,
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/app.module.ts:21]: What are all these imports for?

File: Konteh/front-office-app/src/app/modules/exams/exams.service.ts [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { environment } from "../../enviroments/environment.development";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ExamsService{
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/modules/exams/exams.service.ts:9]: ??

**Review by laslorajda — APPROVED** [review]

**Pull request merged**